tests/support/pretest.js

```diff
@@ -0,0 +1,127 @@
+/*jshint node: true */
+var _ = require('lodash'),
+	/*jshint -W079 */
+	Promise = require('bluebird'),
+	atob = require('atob'),
+	/*jshint +W079 */
+	fs = Promise.promisifyAll(require('fs-extra')),
+	glob = Promise.promisify(require('glob')),
+	path = require('path'),
+	pkg = require('../../package.json')['web-platform-tests'],
+	request = Promise.promisifyAll(require('request')),
+	url = require('url');
+
+var testPath = path.resolve(path.join(__dirname, '..', '..', pkg.path)),
+	token = 'Mjk5ZGQxNTk0ZDA3YTllY2I5YzlmMzRhZWYyOTEyZTQ1MDE2ZDdmNw==',
+	treeUrl = 'https://api.github.com/repos/${owner}/${repo}/git/trees/${sha}';
+
+var harnessFiles = [
+	'testharness.css',
+	'testharness.js',
+	'testharnessreport.js'
+];
+
+var reqOpts = {
+	'headers': {
+		'Authorization': 'token ' + atob(token),
+		'User-Agent': 'request'
+	}
+};
+
+var options = {
+	'raw': function(entry) {
+		return defaultsDeep({
+			'url': entry,
+			'headers': { 'Accept': 'application/vnd.github.v3.raw' }
+		}, reqOpts);
+	},
+	'tree': function(entry) {
+		if (url.parse(entry).hostname !== 'api.github.com') {
+			var parts = entry.match(/([^\/]+)\/([^#]+)#([\s\S]+)/),
+				owner = _.result(parts, 1),
+				repo = _.result(parts, 2),
+				sha = _.result(parts, 3);
+
+			entry = _.template(treeUrl)({ 'owner': owner, 'repo': repo, 'sha': sha });
+		}
+		return defaultsDeep({ 'url': entry, 'json': true }, reqOpts);
+	}
+};
+
+var defaultsDeep = _.partialRight(_.merge, function defaults(objVal, srcVal) {
+	return objVal === undefined ? srcVal : _.merge(objVal, srcVal, defaults);
+});
+
+function getRaw(entry) {
+	return request.getAsync(options.raw(entry))
+		.then(function(res) {
+			return res[1];
+		}, function(err) {
+			throw err;
+		});
+}
+
+function getTree(entry) {
+	return request.getAsync(options.tree(entry))
+		.then(function(res) {
+			return _.result(res, '[1].tree');
+		}, function(err) {
+			throw err;
+		});
+}
+
+module.exports = function() {
+	if (fs.existsSync(testPath)) {
+		return Promise.resolve();
+	}
```

Comment:
I meant if you've already run this task, then later `package.json` is updated to point to a newer SHA, when this task is run again it won't get the updated files. The equivalent is updating a dependency version in `package.json` and then running `npm install` again; you'd get the updated dependency.